packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -117,11 +122,15 @@ class _InspectorReferenceData {
 ///
 /// All methods returning String values return JSON.
 class WidgetInspectorService {
-  WidgetInspectorService._();
+  // This class is intended to be used as a mixin, and should not be
+  // extended directly.
+  factory WidgetInspectorService._() => null;
 
   /// The current [WidgetInspectorService].
   static WidgetInspectorService get instance => _instance;
-  static final WidgetInspectorService _instance = new WidgetInspectorService._();
+  static final WidgetInspectorService _instance = new _WidgetInspectorService();
+
+  bool _debugServiceExtensionsRegistered = false;
```

Comment:
shouldn't this be static?

Reply:
Yes. Changed it to static and cleaned up how the WidgetInspectorService instance was set for
testing purposes so that the asserts don't fail.